### DISCUSSION TITLE
Rootless containers

### DISCUSSION THREAD

**Discussion by shagog** [Jip-Hop/ubernerd#4] [Q&A]
Thanks you for the convenient way to run containers in TrueNAS Scale! I’ve been running several containers with not-Docker 😄  compose for about a week now and it works flawlessly (and makes much more sense than fighting with default ‘apps’ in Scale).

Meanwhile, I’ve been researching the way to run containers in rootless mode, however I feel that my understanding of the nerdctl/containerd is not enough.

I’v been trying to follow the default manual for nerdctl (https://github.com/containerd/nerdctl/blob/main/docs/rootless.md), 
The first thing I encountered is that `$XDG_RUNTIME_DIR` (prerequisite according to https://rootlesscontaine.rs/getting-started/common/login/) variable is not set for admin user in TrueNAS Scale. 

It is possible set it manually, however it’s expected to be` /run/user/$UID `, which directly interferes with the idea not to modify the hosts file system. Also, not sure how will it affect default ‘apps’ .

Have you by any chance researched the possibilities of rootless containers with ubernerd? If so, maybe you could point me towards the proper direction?

**Comment by Jip-Hop** [comment]
I have not really looked into it. But I don't think it has to be a problem to use [the `/run` directory](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s15.html). It contains runtime data inside `tmpfs` and will be gone on reboot anyway. So you're not modifying the (persistent) host root filesystem.

```
admin@truenas:~$ mount | grep /run
tmpfs on /run type tmpfs (rw,nosuid,nodev,noexec,relatime,size=6558692k,mode=755,inode64)
```

Personally I'd just use [`jailmaker`](https://github.com/Jip-Hop/jailmaker) or `ubernerd` to create a persistent rootfull container. Inside this container you can then install `docker`, `podman`or `nerdctl` and run rootless containers as per the usual instructions (although I have not tried this myself).

But if you manage to run rootless containers directly please share your process! 😄
